File: test/fixtures/extractor-repo/src/java/com/example/panels/PanelService.java
package com.example.panels;

import java.net.URI;
import static java.util.Objects.requireNonNull;

public class PanelService {
    private final PanelClient client;

    public PanelService(PanelClient client) {
        this.client = requireNonNull(client);
    }

    public Panel loadPanel(String id) {
        audit(id);
        return client.fetch(URI.create("/panels/" + id));
    }

    private void audit(String id) {
    }

    public record Panel(String id) {
    }

    interface PanelClient {
        Panel fetch(URI uri);
    }

    enum PanelStatus {
        ACTIVE
    }

    public @interface PanelBinding {
    }
}

File: test/fixtures/extractor-repo/src/scala/PanelService.scala
package com.example.panels

import cats.effect.IO
import com.example.panels.client.PanelClient

trait PanelRepository {
  def findPanel(id: String): IO[String]
}

final class PanelService(client: PanelClient) {
  def loadPanel(id: String): IO[String] = {
    client.fetch(id)
  }

  private val cacheName = "panels"
}

object PanelRoutes {
  def route(service: PanelService): String = "panel"
}
